botocove/cove_sessions.py

```diff
@@ -94,9 +99,18 @@ def _cove_session_factory(self, account_id: str) -> CoveSession:
 
         try:
             logger.debug(f"Attempting to assume {role_arn}")
-            creds = self.sts_client.assume_role(
-                RoleArn=role_arn, RoleSessionName=self.role_session_name
-            )["Credentials"]
+            # This calling style avoids a ParamValidationError from botocore.
+            # Policy is optional, but passing None is not allowed.
+            assume_role_args = {
+                k: v
+                for k, v in [
+                    ("RoleArn", role_arn),
+                    ("RoleSessionName", self.role_session_name),
+                    ("Policy", self.policy),
+                ]
+                if v is not None
```

Comment:
Interestingly I'd expect a Python function to allow you to pass it's default arg without complaint so that's a bit of a quirk of boto3?

Interesting problem to try and solve but probably not worth worrying about - all I can think of for testing the third party package behaviour is asserting we get the right kind of exception but I'm not super keen on writing that many tests for every boto3 call!

This solution does break the type system but cleaner than writing some sort of switch statement so we can just squash it for this line :)